src/3d/qgsshadowrenderingframegraph.cpp

```diff
@@ -406,4 +446,18 @@ void QgsShadowRenderingFrameGraph::setSize( QSize s )
   mSize = s;
   mForwardColorTexture->setSize( mSize.width(), mSize.height() );
   mForwardDepthTexture->setSize( mSize.width(), mSize.height() );
+  mRenderCaptureColorTexture->setSize( mSize.width(), mSize.height() );
+  mRenderCaptureDepthTexture->setSize( mSize.width(), mSize.height() );
+  mRenderSurfaceSelector->setExternalRenderTargetSize( mSize );
+}
+
+void QgsShadowRenderingFrameGraph::setRenderCaptureEnabled( bool enabled )
+{
+  if ( enabled == mRenderCaptureEnabled )
+    return;
+  mRenderCaptureEnabled = enabled;
+  if ( mRenderCaptureEnabled )
+    mRenderCaptureTargetSelector->setParent( mPostprocessPassLayerFilter );
+  else
+    mRenderCaptureTargetSelector->setParent( ( Qt3DCore::QNode * ) nullptr );
```

Comment:
Can't we simply call `mRenderCaptureTargetSelector->setEnabled( mRenderCaptureEnabled )` to get the expected result instead of removing parent and then re-assigning it back?